### PR TITLE
Install CSI drivers for k8s 1.23 un-conditionally

### PR DESCRIPTION
**Pull request by ahmedwaleedmalik**
Signed-off-by: Waleed Malik <ahmedwaleedmalik@gmail.com>

**What this PR does / why we need it**:
Install available CSI driver for k8s 1.23 un-conditionally

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Install available CSI driver for k8s 1.23 un-conditionally
```

### CODE REVIEW

**Review by xmudrii — CHANGES_REQUESTED** [review]

File: pkg/addons/ensure.go [hunk 1]
```diff
@@ -413,3 +306,126 @@ func DeleteAddonByName(s *state.State, addonName string) error {
...
+				name: resources.AddonCSIDigitalOcean,
+			},
+		)
+	// Install CSI driver unconditionally
+	case s.Cluster.CloudProvider.Hetzner != nil:
+		addonsToDeploy = append(addonsToDeploy,
+			addonAction{
+				name: resources.AddonCSIHetzner,
+			},
+		)
```
Comment (xmudrii) [inline pkg/addons/ensure.go:352]: Can we also deploy the Nutanix CSI driver here?

```suggestion
		)
	case s.Cluster.CloudProvider.Nutanix != nil:
		addonsToDeploy = append(addonsToDeploy,
			addonAction{
				name: resources.AddonCSINutanix,
			},
		)
```

You would also have to add it here: https://github.com/kubermatic/kubeone/blob/20e645cebb4680fb3b6c3ef22ba82800b5034b6a/pkg/addons/ensure.go#L42

Comment (ahmedwaleedmalik) [reply pkg/addons/ensure.go:353]: Done

**Review by xmudrii — APPROVED** [review]

**Pull request merged**